### ISSUE TITLE
Improve accessibility for radio/checkbox

### ISSUE THREAD

**Issue by nchan0154** [hankchizljaw/boilerform#7]
- Inputs have `visibility: hidden`, which makes them unfocusable so keyboard users cannot tab to them, switch to having item visible but positioned offscreen
- Also need to add CSS for the custom checkbox/radio element so that when it is adjacent to a focused offscreen input, a focus ring appears

I'll create a PR once discussed changes are approved! :)

**Comment by Andy-set-studio** [comment]
Awesome, thank you @nchan0154 🙂

Would something the [visually-hidden helper](https://github.com/hankchizljaw/boilerform/blob/master/assets/scss/_helpers.scss#L11) help for point one? I need to update it because some good feedback landed on the [post I did about #a11y on CSS-Tricks](https://css-tricks.com/small-tweaks-can-make-huge-impact-websites-accessibility/#article-header-id-5). I just completely forgot about this project 🙃

For point two, go ahead. I didn't realise that'd be the case, so thanks for pointing it out! 

I look forward to seeing the PR 🚀

**Comment by nchan0154** [comment]
Great article! I think the key thing for the first point is not having `visibility: hidden` on it so that it can receive focus for keyboard users, so slightly different needs to screenreaders there, I will probably use the snippet in the article + positioning the element with `left: 9999px` or something like that just in case of any old browser weirdness.

**Comment by Andy-set-studio** [comment]
Sweet, sounds good! 

I think the helper might do the job for you with old browsers. That's what the various `clip` properties are supposed to cover, but I'll leave it to your judgment 🙂

Thanks again for offering to fix this!

**Comment by Andy-set-studio** [comment]
This is looking great, @nchan0154! 

Very elegantly done 🙂

My only concern is the visually hidden helper. What do you reckon to us implementing the following in this PR too?

```scss
%visually-hidden { 
    display: block;
    height: 1px;
    width: 1px;
    overflow: hidden;
    clip: rect(1px 1px 1px 1px);
    clip: rect(1px, 1px, 1px, 1px);
    clip-path: inset(1px);
    visibility: hidden;
    white-space: nowrap;
    position: absolute;
}
```

I personally just think that adding a width and height back in and a better clip system would resolve some of the feedback I got in the post.

It'd be great to get your opinion though!

**Comment by nchan0154** [comment]
It looks fine, but we can't have visibility: hidden on it because then certain elements wouldn't be [focusable](https://developer.mozilla.org/en-US/docs/Web/CSS/visibility)! Other than that though, definitely works. :)

**Comment by Andy-set-studio** [comment]
Oh @nchan0154, I'm sorry. I'm such a dufus! 

Yeh we'll definitely remove that property.

I'll blame it on a long day yesterday or something...🙃

I'll add that helper change into this PR and merge it then 🎉

**Comment by Andy-set-studio** [comment]
Any contributions will always be welcomed! Thank you 🙂

![High five](https://media.giphy.com/media/HX3lSnGXZnaWk/giphy.gif)
